### PR TITLE
tenant support to v1.x

### PR DESCRIPTION
**Pull request by juhanakristian**
Continuing the work of @Twister42 and @HegarGarcia

Fixes some issues mentioned in  #429 comments

The changes were originally made to `main` but this PR is made against v1.x branch

### CODE REVIEW

File: src/__tests__/firebaseAdmin.test.js [hunk 1]
```diff
@@ -1111,5 +1113,40 @@ describe('getCustomIdAndRefreshTokens', () => {
       'A `fetch` global is required when using next-firebase-auth. See documentation on setting up a `fetch` polyfill.'
     )
   })
+
+  it('changes auth instace if there is a tenant', async () => {
```
Comment (kmjennison) [inline src/__tests__/firebaseAdmin.test.js]: Typo: "instace"

File: src/__tests__/initFirebaseClientSDK.test.js [hunk 2]
```diff
@@ -115,4 +115,37 @@ describe('initFirebaseClientSDK', () => {
...
+    expect.assertions(1)
+    const mockConfig = createMockConfig()
+    setConfig({
+      ...mockConfig,
+      firebaseClientInitConfig: {
+        ...mockConfig.firebaseClientInitConfig,
+        tenantId: 'my-tenant-id',
+      },
+    })
+    // firebase.apps = []
```
Comment (kmjennison) [inline src/__tests__/initFirebaseClientSDK.test.js]: Remove unused comment

File: src/__tests__/initFirebaseClientSDK.test.js [hunk 3]
```diff
@@ -115,4 +115,37 @@ describe('initFirebaseClientSDK', () => {
...
+      },
+      set tenantId(value) {
+        this.mockTenantId = value
+      },
+    }
+
+    getApps.mockReturnValue([])
+    getAuth.mockReturnValue(mockAuth)
+    // Mocking the firebase.auth() function
+    // jest.spyOn(firebase, 'auth').mockReturnValue(mockAuth)
```
Comment (kmjennison) [inline src/__tests__/initFirebaseClientSDK.test.js]: Remove unused comment

File: index.d.ts [hunk 4]
```diff
@@ -97,6 +98,7 @@ interface InitConfig {
     messagingSenderId?: string
     // "G-MEASUREMENT_ID"
     measurementId?: string
+    tenantId?: string
```
Comment (kmjennison) [inline index.d.ts]: Let's move this out of `firebaseClientInitConfig` and onto the top level. For ease of use, I think it's useful to leave `firebaseClientInitConfig` as purely the inputs to `initializeApp`.

**Review by kmjennison — CHANGES_REQUESTED** [review]
This looks great, thank you for the PR! Just one minor change on the config and a few small comments, then good to merge.

File: src/__tests__/firebaseAdmin.test.js (hunk 1 above)
Comment (juhanakristian) [reply src/__tests__/firebaseAdmin.test.js]: Thanks, fixed now

File: src/__tests__/initFirebaseClientSDK.test.js (hunk 2 above)
Comment (juhanakristian) [reply src/__tests__/initFirebaseClientSDK.test.js]: Fixed

File: src/__tests__/initFirebaseClientSDK.test.js (hunk 3 above)
Comment (juhanakristian) [reply src/__tests__/initFirebaseClientSDK.test.js]: Fixed now

File: index.d.ts (hunk 4 above)
Comment (juhanakristian) [reply index.d.ts]: This has now been fixed

**Comment by juhanakristian** [comment]
The review issues have been fixed but I'll need to fix the issues from the TypeScript conversion

**Comment by kmjennison** [comment]
Appreciate it! Sorry for blowing up your PR with all the changes!

**Comment by juhanakristian** [comment]
The conflicts have been resolved now but it looks like bundle size check is failing in the pipeline. Anything I should do about that?

File: src/__tests__/firebaseAdmin.test.ts [hunk 5]
```diff
@@ -944,6 +944,7 @@ describe('getCustomIdAndRefreshTokens', () => {
     })
     const mockUserRecord = createMockFirebaseUserRecord()
     const mockFirebaseUser = createMockFirebaseUserAdminSDK()
+    console.log('mockFirebaseUser', mockFirebaseUser)
```
Comment (kmjennison) [inline src/__tests__/firebaseAdmin.test.ts]: Please remove console log.

**Review by kmjennison — CHANGES_REQUESTED** [review]
Looks great. Just one minor edit to remove a console log, then good to go! I bumped up the bundle size limit, so no worries on that.

File: src/__tests__/firebaseAdmin.test.ts (hunk 5 above)
Comment (kmjennison) [inline src/__tests__/firebaseAdmin.test.ts]:
```suggestion
```

**Review by kmjennison — APPROVED** [review]

**Comment by juhanakristian** [comment]
Thanks for making the last fix and merging the PR  👍

**Comment by kmjennison** [comment]
Of course! Thank you for your work on this!

**Pull request merged**